Synergy/Frontend/src/app/services/api-requests/teams/get-teams.service.ts

```diff
@@ -0,0 +1,34 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from "@angular/common/http";
+import {API_ROOT} from "../../../../constants/constants";
+import {map} from "rxjs";
+import {TeamsActions} from "@app/store/actions/teams.action";
+import {Store} from "@ngxs/store";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GetTeamsService {
+
+  constructor(private http: HttpClient, private store: Store) {
+  }
+
+  getTeams(teamIds: string[]) {
+    const url = `${API_ROOT}Teams/get-user-teams`;
+    if (teamIds.length === 0)
+      return;
+    return this.http.post(url, teamIds);
+  }
+
+  getTeamsAndUpdateStore(teamIds: string[]) {
+
+    const teams = this.getTeams(teamIds)
+    if (teams) {
```

Comment:
Services should return the observable not handling it itself, so that it's consumer can decide how to use it

Reply:
I did this because I was writing the same logic for the handling the observable in multiple components, so I thought I would put it as a separate method in the service being mindful of DRY. Is there a better way to do this?